src/tools/tidy/src/main.rs

```diff
@@ -110,10 +110,7 @@ fn main() {
             check!(bins, &root_path);
         }
 
-        check!(style, &src_path);
-        check!(style, &tests_path);
-        check!(style, &compiler_path);
-        check!(style, &library_path);
```

Comment:
Note that that applying the check separately we're getting parallelism. I suspect that this will make tidy slower since `style` is one of the longer-running tasks.

Can you benchmark this?

Reply:
It doesn't *seem* to be any worse (no numbers). On the roadmap for `tidy` is file-level parallelism in `walk` which would alleviate any possible perf regressions.

Reply:
Can you measure the performance? Something like hyperfine may help there.

Reply:
Sorry, I'm on holiday right now and only have access to my laptop (which can barely run `x test tidy` in the first place lol). I will note that it's not noticeably different but I have no benchmarks to confirm that at the micro-scale.

Reply:
Without this patch: 2.2 seconds, with this patch: 2.6 seconds (results consistent across two systems, one with macOS the other with Linux). I _personally_ don't really pay super close attention to tidy test times, but I believe folks are running it in e.g. commit hooks so this might be significant. @the8472 do you have an opinion on this?

Reply:
So the intent here is to just get `README.md` under tidy control? If so then spending 0.4 seconds more on just that seems a bit much. And yes I run tidy while rebasing so the extra wait can be a bit annoying.

Would it work to add `README.md` as additional root point for the traversal?

> On the roadmap for tidy is file-level parallelism in walk which would alleviate any possible perf regressions.

If it's on the radar I won't object strongly to have this merged but if we can achieve the goal without the loss it would still be preferable.

Reply:
> So the intent here is to just get README.md under tidy control?

Well, more than just `README.md`, all the other files from the root directory which aren't in the well-defined directories (for instance the `RELEASES.md` formatting issue was caught here; as well as something which I put in another PR I think).

Reply:
Anyway, if we can somehow avoid putting everything into a single task instead add a way to only capture the additional files at the root level (e.g. globbing `/*` but not `/**/*`?) then that'd help with the performance issue.